src/metabase/query_processor/middleware/add_implicit_joins.clj

```diff
@@ -144,14 +142,6 @@
            (let [previous-stage (get-in query previous-path)]
              (visible-joins query previous-path previous-stage)))]))
 
-(mu/defn- distinct-fields :- [:or ::qp.schema/xform ::lib.schema/fields]
```

Comment:
Simplified this middleware a lot; now it can use `lib/normalize` to make things distinct as needed.